cfgmgr/natmgr.cpp

```diff
@@ -0,0 +1,6391 @@
+        m_appTwiceNatTableProducer(appDb, APP_NAT_TWICE_TABLE_NAME),
+        m_appTwiceNaptTableProducer(appDb, APP_NAPT_TWICE_TABLE_NAME),
+        m_appNatGlobalTableProducer(appDb, APP_NAT_GLOBAL_TABLE_NAME),
+        m_appNaptPoolIpTable(appDb, APP_NAPT_POOL_IP_TABLE_NAME)
+{
+    SWSS_LOG_ENTER();
+  
+    SWSS_LOG_INFO("NatMgr Constructor ..!");
+
+    /* Set the Admin mode to disabled */
+    natAdminMode = DISABLED;
+
+    /* Set NAT default timeout as 600 seconds */
+    m_natTimeout = NAT_TIMEOUT_DEFAULT;
+    
+    /* Set NAT default tcp timeout as 86400 seconds (1 Day) */
+    m_natTcpTimeout = NAT_TCP_TIMEOUT_DEFAULT;
+
+    /* Set NAT default udp timeout as 300 seconds */
+    m_natUdpTimeout = NAT_UDP_TIMEOUT_DEFAULT;
+
+    /* Clean the NAT iptables */
+    std::string res;
+    const std::string cmds = std::string("") + IPTABLES_CMD + " -F -t nat ";
+    if (swss::exec(cmds, res))
+    {
+        SWSS_LOG_ERROR("Command '%s' failed", cmds.c_str());
+    }
+
+    flushNotifier = std::make_shared<swss::NotificationProducer>(appDb, "FLUSHNATREQUEST");
+}
+
+/* To check the given port is State Ok or not */
+bool NatMgr::isPortStateOk(const string &port)
+{
+    vector<FieldValueTuple> temp;
+
+    if (!port.compare(0, strlen(VLAN_PREFIX), VLAN_PREFIX))
+    {
+        if (m_stateVlanTable.get(port, temp))
+        {
+            SWSS_LOG_INFO("Vlan %s is ready", port.c_str());
+            return true;
+        }
+        SWSS_LOG_INFO("Vlan %s is not yet ready", port.c_str());
+    }
+    else if (!port.compare(0, strlen(LAG_PREFIX), LAG_PREFIX))
+    {
+        if (m_stateLagTable.get(port, temp))
+        {
+            SWSS_LOG_INFO("Lag %s is ready", port.c_str());
+            return true;
+        }
+        SWSS_LOG_INFO("Lag %s is not yet ready", port.c_str());
+    }
+    else if (!port.compare(0, strlen(ETHERNET_PREFIX), ETHERNET_PREFIX))
+    {
+        if (m_statePortTable.get(port, temp))
+        {
+            SWSS_LOG_INFO("Port %s is ready", port.c_str());
+            return true;
+        }
+        SWSS_LOG_INFO("Port %s is not yet ready", port.c_str());
+    }
+    else
+    {
+        SWSS_LOG_ERROR("Invalid Port %s ", port.c_str());
+    }
+    return false;
+}
+
+/* To check the give interface is State Ok or not */
+bool NatMgr::isIntfStateOk(const string &interface)
+{
+    vector<FieldValueTuple> temp;
+
+    if (m_stateInterfaceTable.get(interface, temp))
+    {
+        SWSS_LOG_INFO("Interface %s is ready", interface.c_str());
+        return true;
+    }
+
+    SWSS_LOG_INFO("Interface %s is not yet ready", interface.c_str());
+    return false;
+}
+
+/* To check the nat fetaure is enabled or not */
+bool NatMgr::isNatEnabled(void)
+{
+    if (natAdminMode == ENABLED)
+    {
+        return true;
+    }
+
+    return false;
+}
+
+/* To check the give global_ip is withing the Prefix subnet or not */
+bool NatMgr::isGlobalIpMatching(const string &prefix, const string &global_ip)
```

Comment:
This would fit as a method in IpPrefix class

Reply:
Yes, will move it to IpPrefix class in next patch.